src/addIndex.js

```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
```

Comment:
> (a ..., Int, [a] -> b)

i think we should start wrapping these for clarity. For instance, this is not some `a`'s, an `Int`, and a function from `[a] -> b`. clearer IMO would be:

`((a ..., Int, [a]) -> b)`